### PR TITLE
Implement `find_full_path` within `ephys` modules

### PR DESCRIPTION
**Pull request by kabilar**
- And moved functions to `element-data-loader`.
- All `workflow-array-ephys` pytests passed.
- Rename `element-data-loader` to `element-interface`.

### CODE REVIEW

File: element_array_ephys/ephys.py [hunk 1]
```diff
@@ -442,31 +449,34 @@ class Curation(dj.Manual):
...
     curation_note='': varchar(2000)  
     """
 
     def create1_from_clustering_task(self, key, curation_note=''):
         """
-        A convenient function to create a new corresponding "Curation"
-         for a particular "ClusteringTask"
+        A function to create a new corresponding "Curation" for a particular 
+        "ClusteringTask", which assumes that no curation was performed on the 
+        dataset
```
Comment (ttngu207) [inline element_array_ephys/ephys.py:462]: I don't think we should explicitly say `which assumes that no curation was performed on the dataset` here. 
Technically users can still use this convenient function to create a new entry to this table using info from the Clustering Task table, for a ***new*** curation that they've done (e.g. they just put the new curated outputs in the same previous directory, overwriting the old results)

**Review by ttngu207 — APPROVED** [review]
Looks good, if anything I'd just suggest a minor change:
```
from element_data_loader.utils import find_root_directory, find_full_path, dict_to_uuid
```
just to cut down the `element_data_loader.utils.` everywhere.

But I don't have a strong preference on this

File: element_array_ephys/ephys.py (hunk 1 above)
Comment (kabilar) [reply element_array_ephys/ephys.py]: Thanks for clarifying.  I removed this phrase.

**Comment by kabilar** [comment]
Thanks @ttngu207.  Changes implemented.  Ready for review and merge.

**Comment by kabilar** [comment]
@ttngu207 In this pull request, I also tested your changes for the case of a missing `fileTimeSecs` in the SpikeGLX meta file.

File: element_array_ephys/ephys.py [hunk 2]
```diff
@@ -140,14 +141,16 @@ class EphysFile(dj.Part):
...
         inserted_probe_serial_number = (ProbeInsertion * probe.Probe & key).fetch1('probe')
 
         # search session dir and determine acquisition software
         for ephys_pattern, ephys_acq_type in zip(['*.ap.meta', '*.oebin'],
                                                  ['SpikeGLX', 'Open Ephys']):
-            ephys_meta_filepaths = [fp for fp in sess_dir.rglob(ephys_pattern)]
+            ephys_meta_filepaths = [fp for fp in session_dir.rglob(ephys_pattern)]
             if ephys_meta_filepaths:
                 acq_software = ephys_acq_type
                 break
```
Comment (CBroz1) [inline element_array_ephys/ephys.py:156]: Suggestion for L#160
```
                f' Neither SpikeGLX nor Open Ephys recording files found'
                f' in {session_dir}'
```

File: element_array_ephys/ephys_chronic.py [hunk 3]
```diff
@@ -143,14 +143,15 @@ class EphysFile(dj.Part):
...
         inserted_probe_serial_number = (ProbeInsertion * probe.Probe & key).fetch1('probe')
 
         # search session dir and determine acquisition software
         for ephys_pattern, ephys_acq_type in zip(['*.ap.meta', '*.oebin'],
                                                  ['SpikeGLX', 'Open Ephys']):
-            ephys_meta_filepaths = [fp for fp in sess_dir.rglob(ephys_pattern)]
+            ephys_meta_filepaths = [fp for fp in session_dir.rglob(ephys_pattern)]
             if ephys_meta_filepaths:
                 acq_software = ephys_acq_type
                 break
```
Comment (CBroz1) [inline element_array_ephys/ephys_chronic.py:157]: Suggestion for L#161
```
                f' Neither SpikeGLX nor Open Ephys recording files found'
                f' in {session_dir}'
```

File: element_array_ephys/readers/openephys.py [hunk 4]
```diff
@@ -31,13 +31,13 @@
 class OpenEphys:
 
     def __init__(self, experiment_dir):
```
Comment (CBroz1) [inline element_array_ephys/readers/openephys.py:33]: Smart to do this in the class. I instead made adjustments [on the call](https://github.com/datajoint/element-array-ephys/pull/41/files#diff-3a2e32e7f77959667b86eb13f37a6fffbce93cac4e75e8776ebee7ba8f7123b3R197). I like this solution better

**Review by CBroz1 — APPROVED** [review]

File: element_array_ephys/ephys.py [hunk 5]
```diff
@@ -187,12 +190,13 @@ def make(self, key):
...
+                                           meta_filepath)
             self.EphysFile.insert1({
                 **key,
                 'file_path': meta_filepath.relative_to(root_dir).as_posix()})
         elif acq_software == 'Open Ephys':
-            dataset = openephys.OpenEphys(sess_dir)
+            dataset = openephys.OpenEphys(session_dir)
             for serial_number, probe_data in dataset.probes.items():
                 if str(serial_number) == inserted_probe_serial_number:
                     break
```
Comment (dimitri-yatsenko) [inline element_array_ephys/ephys.py:202]:
```suggestion
            for serial_number in dataset.probes:
                if str(serial_number) == inserted_probe_serial_number:
                    break
```

File: README.md [hunk 6]
```diff
@@ -45,14 +44,24 @@ This ephys element features automatic ingestion for spike sorting results from t
...
+    ```
+
++ Upgrade `element-array-ephys` previously installed with `pip`
+    ```
+    pip install --upgrade element-array-ephys
+    ```
+
++ Install `element-data-loader`
+
+    + `element-data-loader` is a dependency of `element-array-ephys`, however it is not contained within `requirements.txt`.
```
Comment (CBroz1) [inline README.md:60]: Why not add `element-data-loader` to requirements? If we're moving to the convention of relying on the utilities on multiple root directories across elements, we might avoid this extra installation step by including it as a requirement.

Comment (kabilar) [reply README.md:64]: Good question @CBroz1 .  A bit of a long explanation.

We have configured `element-data-loader` to allow for installation of optional packages that are not on PyPI, such as CaImAn and scanreader.  This is done using the `extras_require` feature in [setup.py](https://github.com/datajoint/element-data-loader/blob/main/setup.py#L36).  The user would follow these [installation instructions](https://github.com/datajoint/element-data-loader#installation).

Subsequently, from what I understand about PyPI, if a repository has dependencies in `requirements.txt` or `setup.py` that are not on PyPI, then that repository cannot be published to PyPI.  Thus in this case `element-data-loader` cannot be published to PyPI.  And likewise, if we put `element-data-loader` within the `requirements.txt` of `element-array-ephys`, then `element-array-ephys` also would not be able to be published to PyPI.

Given our updated strategy with `element-data-loader` to store the common utility functions among the `elements` and to employ the `extras_require` feature as described above, we have debated not publishing the `elements` on PyPI.  (Note: `element-data-loader` will likely be renamed in the future.)

File: element_array_ephys/readers/openephys.py (hunk 4 above)
Comment (kabilar) [reply element_array_ephys/readers/openephys.py:33]: Hi @CBroz1, unless I am mistaken, I think we are actually implementing this the same way.

Comment (CBroz1) [reply element_array_ephys/readers/openephys.py:33]: No, sorry, I was mistaken. We both switched to using the full path within the make function.

File: README.md (hunk 6 above)
Comment (CBroz1) [reply README.md:64]: Thanks for the explanation!

File: element_array_ephys/ephys.py (hunk 5 above)
Comment (kabilar) [reply element_array_ephys/ephys.py:203]: The variable `probe_data` is used in the [next section](https://github.com/kabilar/element-array-ephys/blob/ce6adf1c6a7409ecd98907f3a3a5c6d50f0c10bb/element_array_ephys/ephys.py#L208-L232) after the `for` loop, so we would need to keep this statement as is.

Comment (dimitri-yatsenko) [reply element_array_ephys/ephys.py:203]: Then I would get `probe_data` after the loop
```suggestion
            for serial_number in dataset.probes:
                if str(serial_number) == inserted_probe_serial_number:
                    probe_data = dataset.probes[serial_number]
                    break
```

**Comment by kabilar** [comment]
Added
- Rename `element-data-loader` to `element-interface`

**Review by dimitri-yatsenko — APPROVED** [review]

**Pull request merged**